include/supernode.h

```diff
@@ -62,18 +62,20 @@ class Supernode {
 
   Supernode(const Supernode& s);
 public:
-  static Supernode* makeSupernode(uint64_t n, uint64_t seed);
-  static Supernode* makeSupernode(uint64_t n, uint64_t seed, std::istream &binary_in);
-
   /**
-   * Makes a supernode at the provided location (and does no error checking).
-   * @param loc     the memory location to put the supernode.
+   * Supernode construtors
    * @param n       the total number of nodes in the graph.
    * @param seed    the (fixed) seed value passed to each supernode.
-   * @return        a pointer to loc, the location of the supernode.
+   * @param loc     (Optional) the memory location to put the supernode.
+   * @return        a pointer to the newly created supernode object
    */
-  static Supernode* makeSupernode(void* loc, uint64_t n, uint64_t seed);
-  static Supernode* makeSupernode(const Supernode& s);
+  static Supernode* makeSupernode(uint64_t n, long seed, void *loc = malloc(bytes_size));
```

Comment:
could this ever be called when `bytes_size` is uninitialized?

Reply:
Currently you have to call the configure function first. I'll change that.

Reply:
So yes

Reply:
not ideal, but i guess we can let it go for now. can you put up an issue for this?